src/index/update.rs

```diff
@@ -337,10 +356,13 @@ pub fn populate(db: &DB) -> Result<()> {
 	});
 
 	{
-		let mut updater = IndexUpdater::new(album_art_pattern, directory_sender, song_sender)?;
-		for target in mount_points.values() {
-			updater.populate_directory(None, target.as_path())?;
-		}
+		let updater = IndexUpdater::new(album_art_pattern, directory_sender, song_sender)?;
+		let mount_points = mount_points.values().collect::<Vec<_>>();
+		mount_points
+			.iter()
+			.par_bridge()
+			.map(|target| updater.populate_directory(None, target.as_path()))
+			.collect::<Result<()>>()?;
```

Comment:
This looks like the same `collect` magic! Does something `Into<Result>` for `Iter<Result>`?

Reply:
Same as above, it returns an error if one of the mounts failed.